cdci_data_analysis/pytest_fixtures.py

```diff
@@ -730,11 +732,57 @@ def dispatcher_fetch_dummy_products(dummy_product_pack: str, reuse=False):
     open(dispatcher_dummy_product_pack_state_fn, "w").write("%s"%time.time())
 
 
+def clone_gitlab_repo(repository_url, repo_dir=None, gitlab_token_name=None, gitlab_token=None, branch_name=None):
+    if repo_dir is None:
+        repo_dir = get_repo_name(repository_url)
+
+    if branch_name is None:
+        branch_name = 'master'
+
+    url_parsed = urlparse(repository_url)
+
+    if gitlab_token_name is not None and gitlab_token is not None:
+        url_parsed = url_parsed._replace(netloc=f'{gitlab_token_name}:{gitlab_token}@{url_parsed.hostname}')
+
+    repo = Repo.clone_from(url_parsed.geturl(), repo_dir, branch=branch_name)
```

Comment:
most if not all of this would work with test on a filesystem repo

Reply:
next step!